app/ante/cosmos/context.go

```diff
@@ -0,0 +1,91 @@
+package cosmos
+
+import (
+	"fmt"
+
+	anteutils "github.com/ExocoreNetwork/exocore/app/ante/utils"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/migrations/legacytx"
+	evmostypes "github.com/evmos/evmos/v14/types"
+)
+
+var _ GasTx = (*legacytx.StdTx)(nil) // assert StdTx implements GasTx
+
+// GasTx defines a Tx with a GetGas() method which is needed to use SetUpContextDecorator
+type GasTx interface {
+	sdk.Tx
+	GetGas() uint64
+}
+
+// SetUpContextDecorator sets the GasMeter in the Context and wraps the next AnteHandler with a defer clause
+// to recover from any downstream OutOfGas panics in the AnteHandler chain to return an error with information
+// on gas provided and gas used.
+// CONTRACT: Must be first decorator in the chain
+// CONTRACT: Tx must implement GasTx interface
+type SetUpContextDecorator struct{}
+
+func NewSetUpContextDecorator() SetUpContextDecorator {
+	return SetUpContextDecorator{}
+}
+
+func (sud SetUpContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
```

Comment:
this  infinite gas transaction OracleCreatePriceTx, we need  add some log to track them in case that resources are misused, and transactions are rampant . how about the frequency of this transaction ?

Reply:
This tx type is limited to create-price and that related gas consume is basically fixed. The bond is:
tx/block: num(validator) * maxNonce / interval(blocks).
eg. we have 21 validators and maxNonce set to 3, interval set to 100 blocks. then the result would be: 0.63tx/block in average. and the execution of tx:create-price is deterministic(which result anticipated gas consumption). Then we can rely on the number of tx to control the resource consumption.

Reply:
We set create-price tx with the highest priority since it's basically a system service. So we limit this tx with nonce limitation not gas consumption(since it will finally fill the txpool with that priority)

Reply:
But if the validator attempts to do a Sybil attack by sending many such transactions, how is it prevented?